src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -108,18 +108,20 @@ export class CommandDetectionCapability implements ICommandDetectionCapability {
 		// On Windows track all cursor movements after the command start sequence
 		if (this._isWindowsPty) {
 			this._commandMarkers.length = 0;
-			this._onCursorMoveListener = this._terminal.onCursorMove(() => {
-				if (this._commandMarkers.length === 0 || this._commandMarkers[this._commandMarkers.length - 1].line !== this._terminal.buffer.active.cursorY) {
-					const marker = this._terminal.registerMarker(0);
-					if (marker) {
-						this._commandMarkers.push(marker);
-					}
-				}
-			});
 			// HACK: Fire command started on the following frame on Windows to allow the cursor
 			// position to update as conpty often prints the sequence on a different line to the
 			// actual line the command started on.
 			timeout(0).then(() => {
+				if (!this._currentCommand.commandExecutedMarker) {
```

Comment:
if conpty does fire command started for the actual line it was started on, are there any consequences of this happening again?

I think we dispose of the placeholder decoration every time register decoration is called, which should prevent duplicates

Reply:
We only want to listen to cursor moves between start and executed, doing it outside that would be a waste as we wouldn't use it.